---
title: 'Code replication -1: single cell analysis for mouse ASC'
date: 2024-01-17
permalink: /posts/2024/01/blog-post-1/
tags:
  - MSC
  - replication
---

## 1. Data source and information
### 1.1 Paper link: 
<https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0242171>

### 1.2 Methods:
barcode sequences were extracted from the read 1 fastq files. The read 2 fastq files, which included each cell mRNA, were directly aligned to Refseq transcript sequences.

### 1.3 Samples:
 Male mice of the C57BL/6NCr strain at 6 and 29 months of age were obtained from the animal facility of TMIG. 

## 2. Process

### Upstream
```
STAR-2.7.10b
```
### 1. download
```
prefetch --option-file SRR_Acc_List.txt --max-size 500G

fastq-dump SRR* --gzip --split-files
```

files:

| file name | read length |
|-------|-------|
| DRR207889_1.fastq | 25 | 
| DRR207889_2.fastq | 59 |
| DRR207890_1.fastq | 25 |
| DRR207890_2.fastq | 59 |

### 2. get barcodes
```
cat DRR207889_1.fastq | sed -n '2~4p' > wl_mid.txt
cut -c 1-12 wl_mid.txt > whitelist.txt
sort whitelist.txt | uniq > mouse_ASC_29months_barcode.txt 
```
### 3. Alignment
```
#generate genome
STAR --runThreadN 8 \
    --runMode genomeGenerate \
    --genomeDir ./STAR_index_GRCm38_gtf_new \
    --genomeFastaFiles ./Mus_musculus.GRCm38.97.dna.primary_assembly.fa \
    --sjdbGTFfile ./Mus_musculus.GRCm38.98.gtf
```

```
STAR --runThreadN 16 \
--soloType CB_UMI_Simple \
--genomeDir /home/zje610/workspace/refData/ensemble/STAR_index_GRCm38_gtf_new \
--soloCBwhitelist ../02_barcode/mouse_ASC_6months_barcode.txt \
--soloBarcodeMate 2 \
--clip5pNbases 0 12 \
--soloCBstart 1 \
--soloCBlen 12 \
--soloUMIstart 1 \
--soloUMIlen 12 \
--outSAMattributes NH HI AS nM CR CY UR UY sM \
--outFilterMultimapNmax 100 \
--winAnchorMultimapNmax 100 \
--outMultimapperOrder Random \
--runRNGseed 777 \
--outSAMmultNmax 1 \
--outSAMtype BAM SortedByCoordinate \
--soloCellFilter EmptyDrops_CR 10000 0.99 10 45000 90000 500 0.01 20000 0.01 10000 \
--outFileNamePrefix mouse_ASC_6months_ \
--readFilesIn ${file}_2.fastq \
${file}_1.fastq
```

### 4. Basic mapping statistics

<style>
  table {
    display: block;
    overflow-x: auto;
    white-space: nowrap;
  }
</style>

| name | Number of input reads | Uniquely mapped reads number | Uniquely mapped reads % | Number of reads mapped to multiple loci | % of reads mapped to multiple loci mapped reads | Number of reads unmapped: too short | % of reads unmapped: too short |
|------|-----------------------|------------------------------|-------------------------|-----------------------------------------|-----------------------------------------------|-------------------------------------|--------------------------------|
|mouse_ASC_6months|156992623|99526246|63.40%|29443786|18.75%|27718389|17.66%|
|mouse_ASC_29months|1162489731|114524499|70.48%|35452911|21.82%|12306366|7.57%|

  
## Downstream
```
R-Seurat
```
### 1. QC
```
#load library
library(Seurat) 
library(dplyr) 
library(patchwork) 

#set dic
setwd("/home/aidi/raw_count")

mouse_ASC_6months.data <- Read10X("./mouse_ASC_6months/")
reateSeuratObject(counts = mouse_ASC_6months.data, project = "mouse_ASC_6months", min.cells = 3, min.features = 100)
mouse_ASC_6months[["percent.mt"]] <- PercentageFeatureSet(mouse_ASC_6months, pattern = "^mt-")
VlnPlot(mouse_ASC_6months, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

mouse_ASC_29months.data <- Read10X("./mouse_ASC_29months/")
reateSeuratObject(counts = mouse_ASC_29months.data, project = "mouse_ASC_29months", min.cells = 3, min.features = 100)
mouse_ASC_29months[["percent.mt"]] <- PercentageFeatureSet(mouse_ASC_29months, pattern = "^mt-")
VlnPlot(mouse_ASC_29months, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
```

before:
![](/attachment/Clipboard_2025-07-28-16-19-37.png)
after:
![](/attachment/Clipboard_2025-07-28-16-20-38.png)
before:
![](/attachment/Clipboard_2025-07-28-16-21-24.png)
after:
![](/attachment/Clipboard_2025-07-28-16-21-33.png)

### 2. Find Variable genes
```
#normalize
mouse_ASC_29months <- NormalizeData(mouse_ASC_29months, normalization.method = "LogNormalize", scale.factor = 10000) 
mouse_ASC_6months <- NormalizeData(mouse_ASC_6months, normalization.method = "LogNormalize", scale.factor = 10000)
```

```
mouse_ASC_29months <- FindVariableFeatures(mouse_ASC_29months, selection.method = "vst", nfeatures = 2000)
top10 <- head(VariableFeatures(mouse_ASC_29months), 10) 
mouse_ASC_29months <- FindVariableFeatures(mouse_ASC_29months, selection.method = "vst", nfeatures = 2000)
top10 <- head(VariableFeatures(mouse_ASC_29months), 10) 
plot1 <- VariableFeaturePlot(mouse_ASC_29months)
plot2 <- LabelPoints(plot = plot1, points = top10, repel = TRUE) 
```
![](/attachment/Clipboard_2025-07-28-16-23-29.png)
![](/attachment/Clipboard_2025-07-28-16-23-34.png)

```
mouse_ASC_6months <- FindVariableFeatures(mouse_ASC_6months, selection.method = "vst", nfeatures = 2000)
top10 <- head(VariableFeatures(mouse_ASC_6months), 10) 
plot1 <- VariableFeaturePlot(mouse_ASC_6months)
plot2 <- LabelPoints(plot = plot1, points = top10, repel = TRUE) 
```

![](/attachment/Clipboard_2025-07-28-16-23-53.png)
![](/attachment/Clipboard_2025-07-28-16-23-59.png)

### 3. PCA
```
mouse_ASC_6months <- ScaleData(mouse_ASC_6months) 

all.genes <- rownames(mouse_ASC_6months) 
mouse_ASC_6months <- ScaleData(mouse_ASC_6months, features = all.genes) 
mouse_ASC_6months <- ScaleData(mouse_ASC_6months, vars.to.regress = "percent.mt")

mouse_ASC_29months <- ScaleData(mouse_ASC_29months) 

all.genes <- rownames(mouse_ASC_29months) 
mouse_ASC_29months <- ScaleData(mouse_ASC_29months, features = all.genes) 
mouse_ASC_29months <- ScaleData(mouse_ASC_29months, vars.to.regress = "percent.mt")
```

```
mouse_ASC_6months <- RunPCA(mouse_ASC_6months, features = VariableFeatures(object = mouse_ASC_6months)) 
mouse_ASC_29months <- RunPCA(mouse_ASC_29months, features = VariableFeatures(object = mouse_ASC_29months)) 
```

```
VizDimLoadings(mouse_ASC_6months, dims = 1:2, reduction = "pca") 
```
![](/attachment/Clipboard_2025-07-28-16-26-19.png)

```
VizDimLoadings(mouse_ASC_29months, dims = 1:2, reduction = "pca") 
```
![](/attachment/Clipboard_2025-07-28-16-26-35.png)

```
ElbowPlot(mouse_ASC_6months)
```
![](/attachment/Clipboard_2025-07-28-16-26-50.png)

```
ElbowPlot(mouse_ASC_29months)
```

![](/attachment/Clipboard_2025-07-28-16-27-04.png)

### 4. Embedding
```
mouse_ASC_29months <- RunTSNE(mouse_ASC_29months,dims = 1:20) 
TSNEPlot(object = mouse_ASC_29months)  + NoLegend()
```
![](/attachment/Clipboard_2025-07-28-16-29-36.png)

```
mouse_ASC_6months <- RunTSNE(mouse_ASC_6months,dims = 1:20)
TSNEPlot(object = mouse_ASC_6months)  + NoLegend()
```
![](/attachment/Clipboard_2025-07-28-16-29-51.png)

```
mouse_ASC_6months <- RunUMAP(mouse_ASC_6months, dims = 1:10)
DimPlot(mouse_ASC_6months, reduction = "umap")  + NoLegend()
```
![](/attachment/Clipboard_2025-07-28-16-30-08.png)

```
mouse_ASC_29months <- RunUMAP(mouse_ASC_29months, dims = 1:10)
DimPlot(mouse_ASC_29months, reduction = "umap")  + NoLegend()
```
![](/attachment/Clipboard_2025-07-28-16-30-41.png)

```
mouse_ASC_6months <- FindNeighbors(object = mouse_ASC_6months, dims = 1:20)
mouse_ASC_6months <- FindClusters(object = mouse_ASC_6months, resolution = 0.5)
TSNEPlot(object = mouse_ASC_6months, label=T)+ NoLegend()
```
![](/attachment/Clipboard_2025-07-28-16-31-06.png)

```
mouse_ASC_29months <- FindNeighbors(object = mouse_ASC_29months, dims = 1:20)
mouse_ASC_29months <- FindClusters(object = mouse_ASC_29months, resolution = 0.5)
TSNEPlot(object = mouse_ASC_29months, label=T)+ NoLegend()
```
![](/attachment/Clipboard_2025-07-28-16-31-23.png)


### 3. Paper conclusion
 ASC markers : Gsn, Cxcl1, Col1a2, Col6a1, Mmp2, Mmp14 
 Age-related changes and pseudotime progression were observed in the genes *Adamts7, Egfr, and Igfbp4*; however, the genes *Snai2, Tgfbr1, Lpl, and Efemp1* did not show significant differences with aging.
